copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/WorkerSinkTask.java

```diff
@@ -108,6 +108,9 @@ public void close() {
     /** Poll for new messages with the given timeout. Should only be invoked by the worker thread. */
     public void poll(long timeoutMs) {
         try {
+            rewind();
+            long backOffMs = context.backoff();
+            timeoutMs = Math.max(timeoutMs, backOffMs);
```

Comment:
I think `Math.min` makes more sense now since the connector is actually not proceed unless the backoff is reached. Also updated the PR to reset the backoff.